tests/infra/clients.py

```diff
@@ -393,6 +403,13 @@ def request(
     def close(self):
         pass
 
+    def extra_headers_count():
+        # curl inserts the following headers in every request
+        #  host: <address>
+        #  user-agent: curl/<version>
+        #  accept: */*
+        return 3
```

Comment:
I don't really like hardcoding this, but can't see a cheap better solution. The actual number of auto-inserted headers depends on whether we pass a body or not, this number is only true for empty bodies like this test uses.